Fix issue with invoke pattern being provided for top navview item

chingucoding opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Added check to ensure we only provide InvokePattern when we are not in top NavigationView
## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Use the syntax "Closes #1234" or "Fixes #5678" so that GitHub will close the issue once the PR is complete. -->
Fixes #2702 
## How Has This Been Tested?
<!--- Please describe how you tested your changes. -->
Tested manually:

![image](https://user-images.githubusercontent.com/16122379/86464875-c0dcd800-bd30-11ea-9604-985744588e76.png)

Update API test
## Screenshots (if appropriate):
<!--- If you are making visual changes to a Control or adding a new Control, please include screenshots showing the result. -->

StephenLPeters commented:
/azp run

StephenLPeters commented:
@chingucoding these test failures look real, the top nav tests appear to be expecting to be able to invoke the item while in top nav.

chingucoding commented:
Right, thanks for the ping! I'll take a look tomorrow.

chingucoding commented:
Tests should run successfully now.

StephenLPeters commented:
/azp run

ranjeshj commented on dev/NavigationView/NavigationViewItemAutomationPeer.cpp:
```diff
@@ -47,7 +47,8 @@ winrt::hstring NavigationViewItemAutomationPeer::GetNameCore()
 {
     if (pattern == winrt::PatternInterface::SelectionItem ||
-        pattern == winrt::PatternInterface::Invoke ||
+        // Only provide Invoke pattern when we are a ListItem i.e. not on top navigation
```
> not on top navigation [](start = 66, length = 22)

From the issue comments, it sounds like we don't want to support invoke on top or left navigation.

chingucoding replied:
Hmm I see. So should NavigationView be "Tab" control in left and top mode then too?

chingucoding replied:
@YuliKl What do you think?

YuliKl replied:
From what I read, Invoke and SelectionItem patterns are supposed to be mutually exclusive. And we need nav view items to support the SelectionItem pattern, so I believe that means removing the Invoke pattern regardless of NavigationView's display mode.

chingucoding replied:
I see, thanks for the clarification. This PR is inline with that requirement in its current state.

StephenLPeters commented:
/azp run

chingucoding commented:
There were test failures due to the fact that some tests expected the NavView items to be invokable. Switched to `Click` and updated the settings invoke test, to test whether clicking works correctly.

StephenLPeters commented:
/azp run

chingucoding commented:
It seems that the CI bugged out here. Can we rerun the CI?

StephenLPeters commented:
/azp run
